### PR TITLE
Docs updates from automation

### PR DESCRIPTION
**Pull request by tas50**
Better examples, improved formatting, better action descriptions.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: data/infra/resources/ruby_block.yaml [hunk 1]
```diff
@@ -43,124 +43,102 @@ properties_list:
...
-  \ to\n      trigger an update (if required) and to then reload the client.rb\n \
-  \     file.\n\n  <!-- -->\n\n  ```ruby\n  directory 'node[:ohai][:plugin_path]'\
-  \ do\n    owner 'chef'\n    recursive true\n  end\n\n  ruby_block 'reload_config'\
-  \ do\n    block do\n      Chef::Config.from_file('/etc/chef/client.rb')\n    end\n\
-  \    action :nothing\n  end\n\n  if node[:ohai].key?(:plugins)\n    node[:ohai][:plugins].each\
-  \ do |plugin|\n      remote_file node[:ohai][:plugin_path] +\"/#{plugin}\" do\n\
-  \        source plugin\n        owner 'chef'\n        notifies :run, 'ruby_block[reload_config]',\
-  \ :immediately\n      end\n    end\n  end\n  ```\n"
+examples: |
+  Re-read configuration data
```
Comment (IanMadd) [inline data/infra/resources/ruby_block.yaml]:
```suggestion
  **Re-read configuration data**
```

File: data/infra/resources/ruby_block.yaml [hunk 2]
```diff
@@ -43,124 +43,102 @@ properties_list:
...
+  ```ruby
+  ruby_block 'reload_client_config' do
+    block do
+      Chef::Config.from_file('/etc/chef/client.rb')
+    end
+    action :run
+  end
+  ```
+
+  Run a block on a particular platform
```
Comment (IanMadd) [inline data/infra/resources/ruby_block.yaml]:
```suggestion
  **Run a block on a particular platform**
```

File: data/infra/resources/ruby_block.yaml [hunk 3]
```diff
@@ -43,124 +43,102 @@ properties_list:
...
+        require 'fileutils'
+        FileUtils.cp "#{node['mysql']['client']['lib_dir']}\\libmysql.dll",
+          node['mysql']['client']['ruby_dir']
+      end
+      not_if { ::File.exist?("#{node['mysql']['client']['ruby_dir']}\\libmysql.dll") }
+    end
+  end
+  ```
+
+  Stash a file in a data bag
```
Comment (IanMadd) [inline data/infra/resources/ruby_block.yaml]:
```suggestion
  **Stash a file in a data bag**
```

File: data/infra/resources/ruby_block.yaml [hunk 4]
```diff
@@ -43,124 +43,102 @@ properties_list:
...
+      item.data_bag('bittorrent')
+      item.raw_data = data
+      item.save
+    end
+    action :nothing
+    subscribes :create, "bittorrent_torrent[#{node['bittorrent']['torrent']}]", :immediately
+  end
+  ```
+
+  Update the /etc/hosts file
```
Comment (IanMadd) [inline data/infra/resources/ruby_block.yaml]:
```suggestion
  **Update the /etc/hosts file**
```

File: data/infra/resources/ruby_block.yaml [hunk 5]
```diff
@@ -43,124 +43,102 @@ properties_list:
...
+    block do
+      rc = Chef::Util::FileEdit.new('/etc/hosts')
+      rc.search_file_replace_line(/^127\.0\.0\.1 localhost$/,
+        '127.0.0.1 #{new_fqdn} #{new_hostname} localhost')
+      rc.write_file
+    end
+  end
+  ```
+
+  Set environment variables
```
Comment (IanMadd) [inline data/infra/resources/ruby_block.yaml]:
```suggestion
  **Set environment variables**
```

File: data/infra/resources/ruby_block.yaml [hunk 6]
```diff
@@ -43,124 +43,102 @@ properties_list:
...
+
+  ruby_block 'initialize' do
+    block do
+      ENV['RBENV_ROOT'] = node[:rbenv][:root]
+      ENV['PATH'] = "#{node[:rbenv][:root]}/bin:#{node[:ruby_build][:bin_path]}:#{ENV['PATH']}"
+    end
+  end
+  ```
+
+  Call methods in a gem
```
Comment (IanMadd) [inline data/infra/resources/ruby_block.yaml]:
```suggestion
  **Call methods in a gem**
```

**Review by IanMadd — COMMENTED** [review]
Bolding, otherwise it looks good.

**Comment by tas50** [comment]
Thanks @IanMadd

**Pull request merged**